docs/design/2022-01-04-integer-shard-index.md

```diff
@@ -0,0 +1,423 @@
+			case ast.EQ:
+				return AddGcColumn4EqCond(sctx, cols, accessesCond, columnValues)
+			case ast.In:
+				return AddGcColumn4InCond(sctx, cols, accessesCond)
+			}
+		}
+	}
+
+	return accessesCond, nil
+}
+```
+
+
+
+##### AddGcColumn4EqCond
+
+Add the `tidb_shard(x) = xxx` prefix for equal access condition.
+
+- function declaration
+
+```go
+// AddGcColumn4EqCond add the `tidb_shard(x) = xxx` prefix for equal condition
+// For param explanation, please refer to the function `AddGcColumnCond`.
+// @retval -  []expression.Expression   the new condtions after adding `tidb_shard() = xxx` prefix
+//            []*valueInfo              the values of every columns in the returned new condtions
+//            error                     if error gernerated, return error
+func AddGcColumn4EqCond(sctx sessionctx.Context,
+	cols []*expression.Column,
+	accessesCond []expression.Expression,
+	columnValues []*valueInfo) ([]expression.Expression, error) 
+```
+
+- Function Process
+  - Traverse every `ColumnValue` in slice `columnValues`, calculate the value of `tidb_shard` by `ColumnValue` and store it to the variable`evaluated`.
+  - Make a new expression `tidb_shard(columnName) = evaluated` and store it to `accessesCond[0]`
+
+
+
+##### AddGcColumn4InCond
+
+ Add the `tidb_shard(x) = xxx` for `IN` condition.
+
+- function declaration
+
+```go
+// AddGcColumn4InCond add the `tidb_shard(x) = xxx` for `IN` condition
+// For param explanation, please refer to the function `AddGcColumnCond`.
+// @retval -  []expression.Expression   the new condtions after adding `tidb_shard() = xxx` prefix
+//            error                     if error gernerated, return error
+func AddGcColumn4InCond(sctx sessionctx.Context,
+	cols []*expression.Column,
+	accessesCond []expression.Expression) ([]expression.Expression, error)
+```
+
+- Function Process
+
+Traverse every `argument` from `In` expression, calculate the `value` of `tidb_shard` with `argument` and make a new expression `tidb_shard(columnName) = value`, then make a `LogicAnd` like `tidb_shard(a) = 8 AND a = 100`. When process the second or above argument , add the `LogicAnd` to the `LogicOR` expression, like `tidb_shard(a) = 8 AND a = 100 OR tidb_shard(a) = 8 AND a = 100`.
+
+
+
+## Drawbacks
+
+Using the shard index to replace the hot index my decrease a little of the performance when the TiDB cluster is small and the pressure of workload is not high, But it can increase the scalability obviously.
+
+
+
+## Alternatives
+
+We can use expression index directly other than as the prefix of a index. e.g. replace `index idx(a)` to `indx idx(expr(a))` , it seems more simpler. But  `expr(a)` and `(a)`  don't correspond one to one, the PointGet plan becomes IndexRangeScan and the selection operation `WEHRE a = xxx` must be executed in the tidb-server or tikv. So the performance may be dropped a lot.
+
+
+
+## Questions
+
+As I know @tiancaiamao is intended to do the shard index more comprehensively.
```

Comment:
If there're links we could add them here.

Reply:
It is mentioned in his [OKR](https://pingcap.feishu.cn/okr/user/6939716303569946143/?okrId=7008277359162934792&type=search).

Reply:
If there's a referencing document we could link it here, by now seems there's no detailed design yet.

Reply:
I'll remove it